server/dbs/postgres_sqlalchemy.py

```diff
@@ -84,6 +83,10 @@ def _isFunctionAllowed(self, funcname):
                 if (func[1] not in ('i', 's') and
                         self._allowedFunctions.get(func[0], False)):
                     self._allowedFunctions[func[0]] = False
+            # Whitelist some functions if they exist
+            for func in funcs:
+                if func[0] in ('pg_sleep', ):
+                    self._allowedFunctions[func[0]] = True
```

Comment:
I want to be able to cancel slow queries.  This has to do with the use case of the user selecting one thing and then selecting another, so the first query could be cancelled as it is no longer needed.  It is difficult to make a query to a immutable function that is both slow enough to ensure we have time to cancel it and is consistent in allowing cancellation (a lot of the PostGIS functions call some C code which doesn't get cancelled during certain phases of the query).  `pg_sleep` is a simple function that is slow and cancellable.  It is not marked as an immutable function in postgres since it has a server side effect, but it is non-volatile.  By white-listing it, I can use it in the cancel-query test.